library/src/main/java/com/chuckerteam/chucker/internal/ui/transaction/TransactionOverviewFragment.kt

```diff
@@ -83,28 +52,33 @@ internal class TransactionOverviewFragment : Fragment() {
             .observe(
                 viewLifecycleOwner,
                 Observer { (transaction, encodeUrl) ->
-                    url.text = transaction?.getFormattedUrl(encodeUrl)
-                    method.text = transaction?.method
-                    protocol.text = transaction?.protocol
-                    status.text = transaction?.status?.toString()
-                    response.text = transaction?.responseSummaryText
-                    if (transaction?.isSsl == true) {
-                        ssl.setText(R.string.chucker_yes)
-                        tlsVersionGroup.visibility = View.VISIBLE
-                        tlsVersion.text = transaction.responseTlsVersion
-
-                        cipherSuiteGroup.visibility = View.VISIBLE
-                        cipherSuiteVersion.text = transaction.responseCipherSuite
-                    } else {
-                        ssl.setText(R.string.chucker_no)
-                    }
-                    requestTime.text = transaction?.requestDateString
-                    responseTime.text = transaction?.responseDateString
-                    duration.text = transaction?.durationString
-                    requestSize.text = transaction?.requestSizeString
-                    responseSize.text = transaction?.responseSizeString
-                    totalSize.text = transaction?.totalSizeString
+                    populateUI(transaction, encodeUrl)
                 }
             )
     }
+
+    private fun populateUI(transaction: HttpTransaction?, encodeUrl: Boolean) {
+        with(overviewBinding) {
+            url.text = transaction?.getFormattedUrl(encodeUrl)
+            method.text = transaction?.method
+            protocol.text = transaction?.protocol
+            status.text = transaction?.status.toString()
+            response.text = transaction?.responseSummaryText
+            ssl.setText(if (transaction?.isSsl == true) R.string.chucker_yes else R.string.chucker_no)
```

Comment:
Shouldn't this show an empty text or hide a field in case `isSsl` is null? Realistically users should never experience since it comes from the request which is processed rather immediately. Just something i noticed.

But maybe this suggests that the DB model is not proper. Maybe it should be split into requests and responses tables and transaction table would refer to those by their IDs. This is definitely not for this PR but maybe it should be discussed in some other ticket? I can create one and describe the issue I see with the current DB model.

Reply:
Yeah, I believe it should be hidden.

As to DB issues - sure, feel free to raise it. I wasn't involved in the current design, so would gladly join if needed.

Reply:
I'm not sure if you forgot about it during CR fixes. Other than that LGTM.

Reply:
Yeah, forgot it. Already fixed. Please approve it if other things are fine.